e2e-tests/src/validators.rs

```diff
@@ -138,9 +138,20 @@ pub async fn prepare_validators<S: SignedConnectionApi + AuthorRpc>(
         }));
     }
 
-    for controller in accounts.controller_raw_keys.iter() {
+    let address_tokens = node.split(':').collect::<Vec<_>>();
+    let ws = address_tokens[0];
+    let ip_address = address_tokens[1];
+    let first_port = address_tokens[2].parse::<usize>()?;
```

Comment:
Eh, this should technically be `u16`, but I see it might be annoying to deal with in the loop below – if you can make this an `u16` without much trouble then please do, otherwise just resolve this comment.